src/core/jpx.js

```diff
@@ -24,20 +24,97 @@ class JpxError extends BaseException {
 }
 
 class JpxImage {
-  static #module = null;
-
-  static decode(data, decoderOptions) {
-    decoderOptions ||= {};
-    this.#module ||= OpenJPEG({ warn });
-    const imageData = this.#module.decode(data, decoderOptions);
-    if (typeof imageData === "string") {
-      throw new JpxError(imageData);
+  static #buffer = null;
+
+  static #handler = null;
+
+  static #instantiationFailed = false;
+
+  static #modulePromise = null;
+
+  static #wasmUrl = null;
+
+  static setOptions({ handler, wasmUrl }) {
+    if (!this.#buffer) {
+      this.#wasmUrl = wasmUrl || null;
+      if (wasmUrl === null) {
+        this.#handler = handler;
+      }
+    }
+  }
+
+  static async #instantiateWasm(imports, successCallback) {
+    try {
+      if (!this.#buffer) {
+        if (this.#wasmUrl !== null) {
+          const response = await fetch(`${this.#wasmUrl}openjpeg.wasm`);
+          this.#buffer = await response.arrayBuffer();
+        } else {
+          this.#buffer = await this.#handler.sendWithPromise("FetchWasm", {
+            filename: "openjpeg.wasm",
+          });
```

Comment:
I don't believe that we have proper test-coverage for this code-path, so I'd suggest adding the following to address that (since it's a fairly small and simple PDF/image).
```diff
diff --git a/test/test_manifest.json b/test/test_manifest.json
index 71751285b..1a7418b9a 100644
--- a/test/test_manifest.json
+++ b/test/test_manifest.json
@@ -6371,6 +6371,14 @@
     "rounds": 1,
     "type": "eq"
   },
+  {
+    "id": "issue19326_main_thread_fetch",
+    "file": "pdfs/issue19326.pdf",
+    "md5": "b4d937017daf439a6318501428e0c6ba",
+    "rounds": 1,
+    "type": "eq",
+    "useWorkerFetch": false
+  },
   {
     "id": "bug1140761",
     "file": "pdfs/bug1140761.pdf",
```